src/main/java/com/epam/ta/reportportal/core/project/impl/UpdateProjectHandlerImpl.java

```diff
@@ -287,10 +289,12 @@ private void assignUser(String name, ProjectRole projectRole, List<String> assig
 		projectUser.setUser(modifyingUser);
 		projectUser.setProject(project);
 		project.getUsers().add(projectUser);
-		aclHandler.permitSharedObjects(project.getId(),
-				name,
-				ProjectRole.PROJECT_MANAGER.higherThan(projectRole) ? BasePermission.READ : BasePermission.ADMINISTRATION
-		);
+
+		List<Permission> permissions = Lists.newArrayList(BasePermission.READ);
+		if (ProjectRole.PROJECT_MANAGER.sameOrLowerThan(projectRole)) {
```

Comment:
You've said that such sequence is used for preventing npe. But inside the method you call projectRole.value. So if you have null here you would have npe anyway. So probably it's better make null check here or somewhere upper and replace with 'projectRole.sameOrHigherThan(PROJECT_MANAGER)'

Reply:
fixed